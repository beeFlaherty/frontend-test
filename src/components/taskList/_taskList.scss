.taskList {
	$parent: &;

	&__listItem {
		position: relative;
		font-size: 1.8rem;
		margin-bottom: $verticalMarginList;
		border-left: 5px solid $textColor;

		&--high {
			background: lighten($priorityHigh, 25%);
			border-color: $priorityHigh;
		}

		&--medium {
			background: lighten($priorityMedium, 25%);
			border-color: $priorityMedium;
		}

		&--low {
			background: lighten($priorityLow, 25%);
			border-color: $priorityLow;
		}
	}

	&__listItem__checkbox {
		position: absolute;
		top: 0;
		left: 0;
		opacity: 0;
		display: inline-block;

		+ #{$parent}__listItem__label::after {
			content: none;
		}

		&:checked + #{$parent}__listItem__label {
			text-decoration: line-through;

			&::after {
				content: "";
			}
		}

		&:focus + #{$parent}__listItem__label {
			&::before {
				box-shadow: 0 0 1px 3px rgba(59, 153, 252, 0.7);
				outline: none;
			}
		}
	}

	&__listItem__label {
		position: relative;
		display: block;
		padding: $verticalPaddingList $horizontalPaddingList $verticalPaddingList 80px;

		&::before {
			content: "";
			position: absolute;
			top: 50%;
			left: 30px;
			margin-top: -8px;
			display: inline-block;
			height: 16px;
			width: 16px;
			border: 1px solid $borderGrey;
			background: #FFF;
		}

		&::after {
			position: absolute;
			top: 50%;
			left: 32px;
			margin-top: -6px;
			content: "";
			display: inline-block;
			height: 6px;
			width: 12px;
			border-left: 2px solid $textColor;
			border-bottom: 2px solid $textColor;
			transform: rotate(-45deg);
		}

		&:hover {
			cursor: pointer;
		}
	}

	&__completed {
		padding: 50px;
		font-size: 1.8rem;
		text-align: center;
		color: lighten($textColor, 10%);
	}

	&__delete {
		position: absolute;
		right: 30px;
		height: 20px;
		width: 20px;
		top: calc(50% - 10px);
		border: 0;
		background: transparent;

		&::before,
		&::after {
			position: absolute;
			left: 10px;
			top: 0;
			content: " ";
			height: 20px;
			width: 3px;
			background-color: $textColor;
		}

		&::before {
			transform: rotate(45deg);
		}

		&::after {
			transform: rotate(-45deg);
		}

		&:hover {
			cursor: pointer;

			&::before,
			&::after {
				background: #FFF;
			}
		}
	}
}
