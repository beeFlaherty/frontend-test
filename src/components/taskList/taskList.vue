<template>
		<section class="taskList" 
			aria-labelledby="taskListLabel">
			<h2 id="taskListLabel" class="sr-only">Todo List</h2>
            <ul class="taskList__list">
  				<li v-for="(task, index) in siteContent.tasks" 
					:key="task.id"
					:class="`taskList__listItem taskList__listItem--${priorityWord(task.importance)}`">
					<input :id="`task-${index}`"
						v-model="task.isDone"
						type="checkbox"
						true-value="true"
						false-value="false"
						class="taskList__listItem__checkbox"
						@change="statusChange($event, task)"> 
    				<label :for="`task-${index}`"
						class="taskList__listItem__label">
						{{ task.title }} 
						<span class="taskList__itemPriority sr-only">priority {{ priorityWord(task.importance) }} </span> 
					</label>
  				</li>
			</ul>
		</section>
</template>

<script>
	import axios from 'axios';
	import { loadProgressBar } from 'axios-progress-bar';

	export default {
		computed: {
			'siteContent'() {
    			return this.$store.state.siteContent
  			}
		},
		methods: {
			priorityWord: function(priority) {
				let priorityText = "not set";

				if (priority === 0){
					priorityText = "high";

				} else if ( priority === 1){
					priorityText = "medium";
				}
				else if ( priority === 2){
					priorityText = "low";
				}

				return priorityText;
			},
			statusChange(event, task){
				loadProgressBar();
				axios.patch(`${this.$root.apiBaseUrl}/${task.id}`, {isDone: task.isDone})
                	.then((response) => {   
                    	console.log(response);
                	})
                	.catch(function(e){
                    	console.log(e);
                	});
			}
		}
	}
	
</script>
