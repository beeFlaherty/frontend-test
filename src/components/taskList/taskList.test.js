/* eslint-disable */
import { mount, shallowMount, createLocalVue  } from 'vue-test-utils'
import taskList from '../../components/taskList/taskList.vue';
import Vuex from 'vuex';
import storeData from '../../js/store';

const localVue = createLocalVue()

localVue.use(Vuex);
const store = new Vuex.Store({
    state: {
		tasks: [
            {"title":"Do the thing",
            "isDone":"false",
            "importance":1},
            {"title":"Do the other thing",
            "isDone":"false",
            "importance":0},
        ],
		showForm: false
	},
})


describe("Task List Component", () => {
    it("newTaskForm sucessfuly rendered using mock data", () => {
      const wrapper = mount(taskList, { 
        store, 
        localVue
      })
        expect(wrapper.isVueInstance()).toBeTruthy()
    })
    test('renders correctly', () => {
        const wrapper = mount(taskList, { 
            store, 
            localVue
          })
        expect(wrapper.element).toMatchSnapshot()
    })
})