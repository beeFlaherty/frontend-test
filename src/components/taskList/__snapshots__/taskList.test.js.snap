// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Form Component renders correctly 1`] = `<!--function (a, b, c, d) { return createElement(vm, a, b, c, d, true); }-->`;

exports[`Task List Component renders correctly 1`] = `<!--function (a, b, c, d) { return createElement(vm, a, b, c, d, true); }-->`;
